src/algorithms/models/vertex.ts

```diff
@@ -20,14 +20,14 @@ class Vertex {
   /*
   * Number of edges directed TO this Vertex
   */
-  inDegree: number = 0;
+  inDegree = 0;
   /**
    * Array containing all neighbors of a Vertex, including associated edge
    */
-  neighbors?: Array<{
+  neighbors?: Set<{
```

Comment:
This doesn't make much sense to me:

1. I have never seen an optional parameter other than function parameter.

2. Even if there could be a vertex with no neighbors, it still should be a required property - if there are no edges to adjacent vertices, better have an empty set than null.

3. I think that the object you specified as a type for that set means nothing, as Set can contain any type by design.

Reply:
This was actually not done by me. @VladSaioc could you have a look at Pepa's comments?